modules/transactions/src/main/java/org/apache/ignite/internal/tx/InternalTransaction.java

```diff
@@ -143,4 +144,17 @@ IgniteBiTuple<ClusterNode, Long> enlist(
      * @return The future.
      */
     CompletableFuture<Void> kill();
+
+    /**
+     * Rolls back the transaction if the timeout is exceeded.
+     *
+     * @return The future.
+     */
+    default CompletableFuture<Void> rollbackTimeoutExceededAsync() {
+        return rollbackAsync();
+    }
+
+    default boolean isTimeoutExceeded() {
```

Comment:
Please specify guaranties of the method in the contract. It's not clear whether the method will return true if logical timeout was exceeded or if it was detected, etc? In other words if transactions with timeout 20 was started at HLC timestamp t(10,10) will isTimeoutExceeded return true if HLC(t30, X)?

Reply:
Added java doc and changed the naming. Does it make sense?

Reply:
Yes.